core/modules/layout/layout.module

```diff
@@ -1221,90 +1258,78 @@ function layout_provides_path($path) {
  */
-function layout_get_layout_by_path($path = NULL, $router_item = NULL) {
+function layout_get_layout_by_path($path = NULL, $router_item = NULL, $is_layout_page = FALSE) {
   if (!isset($router_item)) {
     $router_item = menu_get_item($path);
   }
 
-  $cache = &backdrop_static(__FUNCTION__);
-
-  if (empty($cache[$router_item['path']])) {
-    // If the path is a node preview path, check if a custom layout exists for
-    // nodes and use that instead.
-    // @ todo: This adds very specific code for node previews which too closely
-    // couples Node and Layout modules. This needs to be reverted in a future
-    // commit.
-    $preview_node = NULL;
-    $node_types = node_type_get_types();
-    foreach ($node_types as $type) {
-      $preview_path = 'node/preview/' . $type->type . '/%';
-      if ($router_item['path'] == $preview_path) {
-        $tempstore_id = $router_item['map'][3];
-        $preview_node = node_get_node_tempstore($tempstore_id);
-        break;
-      }
-    }
-    if ($preview_node) {
-      $layouts = layout_load_multiple_by_path('node/%', TRUE);
-    }
-    else {
-      $layouts = layout_load_multiple_by_path($router_item['path'], TRUE);
-    }
-    $selected_layout = NULL;
-    foreach ($layouts as $layout) {
-      // Set context data from the menu path.
-      $menu_contexts = $layout->getContexts(LayoutContext::USAGE_TYPE_MENU);
-
-      foreach ($menu_contexts as $context) {
-        if (isset($context->position)) {
-
-          // If on a node preview page, use the preview from tempstore.
-          // @ todo: This is also specific code for node previews which needs
-          // to be reverted in a future commit.
-          if ($preview_node) {
-            $context->setData($preview_node);
-          }
-          else {
-            if (isset($router_item['map'][$context->position])) {
-              $context_data = $router_item['map'][$context->position];
-              // Check for an object loaded by the menu router.
-              if (is_object($context_data)) {
-                $context->setData($context_data);
-              }
-              else {
-                // If no context is set, load one using the layout info. This
-                // branch is reached when a layout has a context whose data
-                // object is not already loaded into $router_item. The built-in
-                // contexts don't do that, but a contrib module that creates a
-                // LayoutContext can have its load_callback called at this
-                // point.
-                $context_info = layout_get_context_info($context->plugin);
-                if (isset($context_info['load callback'])) {
-                  $context_data = call_user_func_array($context_info['load callback'], array($router_item['original_map'][$context->position]));
-                  $context->setData($context_data);
-                }
+  // Use the static cache, but keyed on the normal path (not system path) since
```

Comment:
Backdrop seems to call that a "normal path" in https://docs.backdropcms.org/api/backdrop/core%21includes%21path.inc/function/backdrop_get_normal_path/1.

Reply:
Hm, yeah I see. "normal" really means "unaliased path". I think the example would still help even if we use "normal path". Something like:

> Use the static cache, but keyed on the normal path (such as node/1) rather than the system path (such as node/%) since client modules might serve up different layouts for the same system path.

Reply:
Since we're tweaking, lets also add a comma before the "since" (too long sentences w/o any punctuation make them harder to read).

Reply:
Examples (and comma — two, in fact, since the "rather" clause is an appositive) added in an update to the PR.